chain/types/execresult.go

```diff
@@ -2,44 +2,41 @@ package types
 
 import (
 	"encoding/json"
-	"fmt"
-	"regexp"
-	"runtime"
-	"strings"
 	"time"
-)
-
-type ExecutionTrace struct {
-	Msg        *Message
-	MsgRct     *MessageReceipt
-	Error      string
-	Duration   time.Duration
-	GasCharges []*GasTrace
 
-	Subcalls []ExecutionTrace
-}
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-state-types/exitcode"
+)
 
 type GasTrace struct {
-	Name string
-
-	Location          []Loc `json:"loc"`
-	TotalGas          int64 `json:"tg"`
-	ComputeGas        int64 `json:"cg"`
-	StorageGas        int64 `json:"sg"`
-	TotalVirtualGas   int64 `json:"vtg"`
-	VirtualComputeGas int64 `json:"vcg"`
-	VirtualStorageGas int64 `json:"vsg"`
+	Name       string
+	TotalGas   int64         `json:"tg"`
+	ComputeGas int64         `json:"cg"`
+	StorageGas int64         `json:"sg"`
+	TimeTaken  time.Duration `json:"tt"`
+}
 
-	TimeTaken time.Duration `json:"tt"`
-	Extra     interface{}   `json:"ex,omitempty"`
+type MessageTrace struct {
+	From   address.Address
+	To     address.Address
+	Value  abi.TokenAmount
+	Method abi.MethodNum
+	Params []byte
+	Codec  uint64
```

Comment:
This specifies the codec of params, yeah? If so can it be given a name/comment that indicates this? e.g. `ParamsCodec`?